addons/stock/models/stock_quant.py

```diff
@@ -788,6 +824,8 @@ def _update_available_quantity(self, product_id, location_id, quantity, lot_id=N
                 'owner_id': owner_id and owner_id.id,
                 'in_date': in_date,
             })
+            if 'quants_cache' in self.env.context:
+                self.env.context['quants_cache'][q.product_id.id, q.location_id.id, q.lot_id.id, q.package_id.id, q.owner_id.id] |= q
```

Comment:
Isn't safer to move this in the `create` ? In case we are missing a create somewhere else (or even in case we add one in the future)